rust/stackable-cockpit/src/xfer/cache.rs

```diff
@@ -168,18 +165,30 @@ impl Cache {
         match &self.backend {
             CacheBackend::Disk { base_path } => {
                 let cache_auto_purge_filepath = base_path.join(CACHE_LAST_AUTO_PURGE_FILEPATH);
+
+                // Only create file if not already present
+                if !fs::try_exists(&cache_auto_purge_filepath)
+                    .await
+                    .context(CacheIoSnafu)?
+                {
+                    write_cache_auto_purge_file(&cache_auto_purge_filepath).await?;
+                }
+
+                // Read and covert timestamp
                 let timestamp = fs::read_to_string(&cache_auto_purge_filepath)
                     .await
                     .context(CacheIoSnafu)?;
 
                 let timestamp: u64 = timestamp.parse().context(ParseIntSnafu)?;
                 let timestamp = UNIX_EPOCH + Duration::from_secs(timestamp);
 
+                // If the auto purge interval elapsed, run purge and write
+                // back the new timestamp
                 if timestamp.elapsed().context(SystemTimeSnafu)? >= self.auto_purge_interval {
                     debug!("Auto-purging outdated cache files");
 
                     self.purge(DeleteFilter::OnlyExpired).await?;
-                    write_cache_auto_purge_file(cache_auto_purge_filepath).await?;
+                    write_cache_auto_purge_file(&cache_auto_purge_filepath).await?;
                 }
```

Comment:
This can be simplified a bit if we say that it's ok to purge when there is no timestamp or there have been time travel shenanigans:

```rust
// Read and covert timestamp
let last_purged_at = match fs::read_to_string(&cache_auto_purge_filepath)
    .await {
        Ok(timestamp) => Some(UNIX_EPOCH + Duration::from_secs(timestamp.parse().context(ParseIntSnafu)?)),
        Err(err) if err.kind() == std::io::ErrorKind::NotFound => None,
        Err(err) => return Err(err).context(CacheIoSnafu),
    }
    .context(CacheIoSnafu)?;

// If the auto purge interval elapsed, run purge and write
// back the new timestamp
if last_purged_at.and_then(|ts| ts.elapsed().ok()).map_or(true, |elapsed| elapsed >= self.auto_purge_interval) {
    debug!("Auto-purging outdated cache files");

    self.purge(DeleteFilter::OnlyExpired).await?;
    write_cache_auto_purge_file(&cache_auto_purge_filepath).await?;
}
```

Reply:
Also thought about this approach. Will add it.

Reply:
Added in a2a4dc1